### PR TITLE
JAMES-2107 New change-set of refactoring

### PR DESCRIPTION
**Pull request by chibenwa**
- Collapse catch block
 - Migrate immutable collections to guava
 - Remove pointless boolean expressions

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/mock/MockMailboxSession.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public User getUser() {
     }
 
     public boolean isOpen() {
-        return close == false;
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/mock/MockMailboxSession.java]: I don't like that change, I find it much easier to read with == false

Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/mock/MockMailboxSession.java]: Fair enough. I reverted the commit.

**Comment by chibenwa** [comment]
```
Failed tests: 
  ReadOnlyUsersLDAPRepositoryTest.testShouldStillWorksAfterRestartingLDAP:148 expected:<[tru]e> but was:<[fals]e>
```

test this please

File: server/container/core/src/main/java/org/apache/james/core/MimeMessageUtil.java [hunk 2]
```diff
@@ -159,24 +159,6 @@ public static void writeMessageBodyTo(MimeMessage message, OutputStream bodyOs)
...
-             * MimeMessage.getInputStream(), such as a
-             * javax.mail.internet.ParseException.
-             * 
-             * The ParseException is precisely one of the reasons why the
-             * getRawInputStream() method exists, so that we can continue to
-             * stream the content, even if we cannot handle it. Again, if we get
-             * an exception, we throw the one that caused us to call
-             * getRawInputStream().
-             */
-            try {
```
Comment (quynhn) [inline server/container/core/src/main/java/org/apache/james/core/MimeMessageUtil.java]: Do we remove this biz out of catching exception?

Comment (chibenwa) [inline server/container/core/src/main/java/org/apache/james/core/MimeMessageUtil.java]: No, let's not disturb too much this code that just demands to break.

This is IDE automatic refactoring, thus safe. I'm not willing to take any risk here.

Comment (quynhn) [inline server/container/core/src/main/java/org/apache/james/core/MimeMessageUtil.java]: OK

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**